add documentation for VCPKG_HASH_ADDITIONAL_FILES (#43)
microsoft/vcpkg-docs: vcpkg/users/triplets.md

Also available as build-type specific `VCPKG_MAKE_CONFIGURE_OPTIONS_DEBUG` and `VCPKG_MAKE_CONFIGURE_OPTIONS_RELEASE` variables.

### VCPKG_HASH_ADDITIONAL_FILES

A cmake list of files to include into the calculation of the abi hash. This is for example required if files defined in `VCPKG_MESON_(NATIVE|CROSS)_FILE(_RELEASE|_DEBUG)?`

> should be part of the calculated abi or if files are included in the triplet/toolchain provided to vcpkg and those files should take part in the abi hash.

Reviewer: Are relative paths supported? What are they relative to?

Author: > Are relative paths supported? What are they relative to?

Theoretical yes. Same behavior as `_wfopen_s` since that is what `get_file_hash` seems to boil down to. So it is relative to `CWD` I would assume. However using absolute paths via `CMAKE_CURRENT_LIST_DIR` is consistent in regards on how `VCPKG_CHAINLOAD_TOOLCHHAIN_FILE` needs to be defined.

> A cmake list of files to include into the calculation of the abi hash. This is for example required if files defined in `VCPKG_MESON_(NATIVE|CROSS)_FILE(_RELEASE|_DEBUG)?`
> should be part of the calculated abi or if files are included in the triplet/toolchain provided to vcpkg and those files should take part in the abi hash.
> 
> This field is optional.

Reviewer: ```suggestion
A list of files to include in the calculation of [package ABI hashes](binarycaching.md#abi-hash).

This field is optional.

This is for example required if files defined in `VCPKG_MESON_(NATIVE|CROSS)_FILE(_RELEASE|_DEBUG)?`
should be part of the calculated abi or if files are included in the triplet/toolchain provided to vcpkg and those files should take part in the abi hash.

Only the contents and order of the files will be considered -- the paths themselves will not be considered.
```

```cmake

> set(VCPKG_HASH_ADDITIONAL_FILES "${CMAKE_CURRENT_LIST_DIR}/file1.cmake;${CMAKE_CURRENT_LIST_DIR}/meson-cross.txt")

Reviewer: ```suggestion
set(VCPKG_HASH_ADDITIONAL_FILES
  "${CMAKE_CURRENT_LIST_DIR}/file1.cmake"
  "${CMAKE_CURRENT_LIST_DIR}/meson-cross.txt"
)
```

Also available as build-type specific `VCPKG_MAKE_CONFIGURE_OPTIONS_DEBUG` and `VCPKG_MAKE_CONFIGURE_OPTIONS_RELEASE` variables.

### VCPKG_HASH_ADDITIONAL_FILES

A list of files to include in the calculation of [package ABI hashes](binarycaching.md#abi-hash).

This field is optional.

> This is for example required if files defined in `VCPKG_MESON_(NATIVE|CROSS)_FILE(_RELEASE|_DEBUG)?`
> should be part of the calculated abi or if files are included in the triplet/toolchain provided to vcpkg and those files should take part in the abi hash.

Reviewer 2: ```suggestion
This is necessary in scenarios where additional files modify the package contents and should be included in the ABI hash calculation, for example:

* Files that are included (via `include(filepath)`) in custom triplets and toolchains.
* Files that are defined in `VCPKG_MESON_(NATIVE|CROSS)_FILE_<CONFIG>`
```

> Only the contents and order of the files will be considered -- the paths themselves will not be considered.

Reviewer 2: ```suggestion
Only the contents and order of the files are considered, the paths of the files do not affect the ABI hash.
```

Also available as build-type specific `VCPKG_MAKE_CONFIGURE_OPTIONS_DEBUG` and `VCPKG_MAKE_CONFIGURE_OPTIONS_RELEASE` variables.

### VCPKG_HASH_ADDITIONAL_FILES

A list of files to include in the calculation of [package ABI hashes](binarycaching.md#abi-hash).

This field is optional.

> This is necessary in scenarios where additional files modify the package contents and should be included in the ABI hash calculation, for example:

Reviewer 2: ```suggestion
Use this when additional files modify the package contents and should be included in the ABI hash calculation, for example:
```
